app/lib/hyrax/work_bag.rb

```diff
@@ -0,0 +1,64 @@
+module Hyrax
+  class WorkBag
+    attr_reader :bag_path, :bag, :work_id
+
+    # @param work_id [Array<String>]
+    # @param time_stamp [String]
+    def initialize(work_ids:, time_stamp:, compression:)
```

Comment:
I don't think you need `compression` in the method params.  It looks like `compression` is never used by this class or the sub-classes.